clouds/aws/create_aws_network_resources.py

```diff
@@ -0,0 +1,745 @@
+        )
+        ec2_assign_name_tag(
+            ec2_client=ec2_client,
+            res_id=nat_gateway["NatGateway"]["NatGatewayId"],
+            tag_name=f"{site_name}-nat-{subnet_type}{count+1}",
+        )
+        nat_gateway_id_list.append(nat_gateway["NatGateway"]["NatGatewayId"])
+    ec2_client.get_waiter("nat_gateway_available").wait(
+        NatGatewayIds=nat_gateway_id_list,
+    )
+    return nat_gateway_id_list
+
+
+def create_public_route_table(ec2_client, site_name, vpc_id) -> list:
+    """
+    Create route table for public routes
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         site_name (str): vpc name
+         vpc_id (str): vpc Id of the associated vpc
+    Return:
+         str (rt_public_id): route table id
+    """
+    LOGGER.info("Creating  Public Route Tables")
+
+    rt_public = ec2_client.create_route_table(VpcId=vpc_id)
+    rt_public_id = rt_public["RouteTable"]["RouteTableId"]
+    rt_details = ec2_client.describe_route_tables(RouteTableIds=[rt_public_id])
+    while True:
+        state = (rt_details["RouteTables"][0])["Routes"][0]["State"]
+        if state == "active":
+            break
+        time.sleep(5)
+    ec2_assign_name_tag(
+        ec2_client=ec2_client,
+        res_id=rt_public_id,
+        tag_name=f"{site_name}-rtb-public",
+    )
+    return rt_public_id
+
+
+def create_private_route_table(ec2_client, subnet_count, vpc_id, site_name) -> list:
+    """
+    Create route table for private routes
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         subent_count (int): no. of subnets
+         vpc_id (str): vpc Id of the associated vpc
+         site_name (str): vpc name
+
+    Return:
+         list: list containing subnet Ids
+    """
+    LOGGER.info("Creating  Private Route Tables")
+    rt_private_ids_list = []
+    for count in range(subnet_count):
+        rt_private = ec2_client.create_route_table(VpcId=vpc_id)
+        rt_private_id = rt_private["RouteTable"]["RouteTableId"]
+
+        rt_details = ec2_client.describe_route_tables(RouteTableIds=[rt_private_id])
+        while True:
+            state = (rt_details["RouteTables"][0])["Routes"][0]["State"]
+            if state == "active":
+                break
+            time.sleep(5)
+        ec2_assign_name_tag(
+            ec2_client=ec2_client,
+            res_id=rt_private["RouteTable"]["RouteTableId"],
+            tag_name=f"{site_name}-rtb-private{count+1}",
+        )
+        rt_private_ids_list.append(rt_private["RouteTable"]["RouteTableId"])
+    return rt_private_ids_list
+
+
+def associate_route_ipv4(ec2_client, route_table_id, internet_gateway_id) -> None:
+    """
+    Send all IPv4 traffic to the internet gateway
+    """
+    LOGGER.info("Creating Routes for  Route Tables")
+    ec2_client.create_route(
+        RouteTableId=route_table_id,
+        DestinationCidrBlock="0.0.0.0/0",
```

Comment:
always the value?

Reply:
yes this pints to the default route , otherwise need to specify each route